CONTRIBUTING.txt

```diff
@@ -0,0 +1,98 @@
+    2. To allow useful contributions to be included as quickly as possible.
+    3. To preserve and improve the code quality while encouraging appropriate experimentation and risk-taking
+    4. To minimize dependence on individual contributors by encouraging a large pool of active contributors
+    5. To be inclusive of many viewpoints and to harness a diverse set of skills
+    6. To provide an encouraging environment where contributors learn and improve their skills
+    7. To protect the free and open nature of the FreeCAD project
+
+2. Fundamentals
+
+    1. FreeCAD uses the git distributed revision control system.
+    2. Source code for the main application and related subprojects is hosted on github.com in the FreeCAD organization, hereafter referred to as 'the platform'.
+    3. Problems are discrete, well-defined limitations or bugs.
+    4. FreeCAD uses the platform's issue-tracking system to track problems
+    5. Patches are sets of code changes that resolve a single problem
+    6. FreeCAD uses the Pull Request workflow for evaluating and accepting patches
+
+
+3. Roles
+    1. "User": An member of the wider FreeCAD community who uses the software.
+    2. "Contributor":  A person who submits a patch that resolves a previously identified problem. Contributors do not have commit access to the repository unless they are also maintainers. Everyone, without distinction or discrimination, SHALL have an equal right to become a Contributor
+    3. "Maintainer": A person who merges patches. Maintainers may or may not be Contributors. Their role is to enforce the process. Maintainers have commit access to the repository.
+    4. "Administrator": Administrators have additional authority to maintain the list of designated maintainers.
+
+4. Licensing, Ownership, and Credit
+    1. FreeCAD uses the Lesser General Public License, version 2, or superior (LGPL2+) for all source code.  Additional details can be found in the LICENSE file.
+    2. All contributions to FreeCAD SHALL use the same license as the project.
+    3. All contributions are owned by their authors unless assigned to another.
+    4. FreeCAD does not have a mandatory copyright assignment policy.
+    5. A Contributor who wishes to be identified in the Credits section of the application "About" dialog is responsible for identifying themselves appropriately by notifying an administrator or maintainer.
+
+5. Contribution Requirements
+
+    1. Patches are submitted in the form of Pull Requests (PR)
+    2. Maintainers and Contributors MUST have a platform account and SHOULD use their real names or a well-known alias. If the platform alias differs from the account alias on the FreeCAD Forum, effort SHOULD be taken to avoid confusion.
+    3. A PR SHOULD be a minimal and accurate answer to exactly one identified and agreed-on problem.
+    4. A PR SHOULD refrain from adding additional dependencies to the FreeCAD project unless no other option is available.
```

Comment:
Proposing a new dependency for FreeCAD needs to be reviewed by the "FreeCAD Architectects", which we may need to have defined in a more formal fashion as well.  It is a non-trivial decision that can trigger other changes in the build and packaging process, thus affecting many other people and even other projects like Debian, Ubuntu, ...
This is also a good time to make sure that the licensing terms of the new dependency are compatible with FreeCAD's licensing terms.
To avoid future issues, we should also collect and store the following:
    Where we got the code and documentation
    Which version it is
    A copy of the licensing terms that applied to the code/docs (the licensing terms may change in the future and you may not be able to go back to the original source in the future)
   And, of course, a copy of the code/docs as we acquired them..
Having this kind of information in an organized form helps to answer questions that might be brought up in the future (even so far as a lawsuit).

Reply:
Good thoughts.  A lot of that should come out during the PR/Review process.  

I added the line to allow us to push back in one narrow case:  
Imagine a code contribution that requires functionality that can be met by either of two libraries.  One is already a dependency but the contributor prefers the other.  In that case, we should be able to refuse to merge unless the contribution is changed to use the existing dependency. 

Beyond that, are you suggesting a change to the text?

Reply:
I agree with the narrow case.
Other than that I think we need an additional process for adding a dependency to be defined,  I think that isn't the focus of this document, so I don't think the text needs to change here.